src/bootstrap/builder.rs

```diff
@@ -377,6 +377,13 @@ impl<'a> Builder<'a> {
         self.ensure(Libdir { compiler, target })
     }
 
+    pub fn sysroot_codegen_backends(&self, compiler: Compiler) -> PathBuf {
+        self.sysroot_libdir(compiler, compiler.host)
+            .parent()
+            .unwrap()
+            .join("codegen-backends")
```

Comment:
Nit pick: `x.parent().unwrap().join("y")` is the same as `x.with_file_name("y")` (unless `x` is root).